src/cnc/flavors/aws/shared/provision/partials/collection/lambda/lambda.tf.j2

```diff
@@ -66,31 +93,43 @@ resource "aws_lambda_function" "{{ service.instance_name }}" {
   source_code_hash = filebase64sha256("./lambda_function_payload.zip")
 
   lifecycle {
-    ignore_changes = [
-      environment,
+    ignore_changes = [ 
       source_code_hash,
       filename
       ]
   }
 
+  dynamic "environment" {
```

Comment:
since we take care of updating environment variables in the lambda deploy, i think it might be best to ignore changes to environment just as it was before. That also removes any need to set environment here and can rm the variable definition from the variables file.